lib/util/common.js

```diff
@@ -11,11 +11,35 @@ function addHook (fastify, pluginOptions) {
   const routes = []
   const sharedSchemasMap = new Map()
 
+  const globalExposeHeadRoutes = fastify.initialConfig.exposeHeadRoutes
+
   fastify.addHook('onRoute', (routeOptions) => {
-    // we need to skip HEAD route
-    // since it may automatically added by fastify
-    // and will conflict with the existing route
-    if (routeOptions.method === 'HEAD') return
+    // fastify will generate HEAD routes if either the global `exposeHeadRoutes`
+    // (default behavior since https://github.com/fastify/fastify/pull/2826) or
+    // `exposeHeadRoute` on route level flag is `true`. If two routes with
+    // operationId are added to the swagger object, is is no longer valid.
+    // therefore we suffix the operationId with `-head`.
+    const hasRouteExposeHeadRouteFlag = routeOptions.exposeHeadRoute != null
+    const exposesHeads = hasRouteExposeHeadRouteFlag
+      ? routeOptions.exposeHeadRoute
+      : globalExposeHeadRoutes
+
+    if (
+      routeOptions.method === 'HEAD' &&
+      exposesHeads === true &&
+      routeOptions.schema !== undefined &&
+      routeOptions.schema.operationId !== undefined
```

Comment:
```suggestion
      routeOptions.schema?.operationId !== undefined
```

Reply:
eslint wont allow me to use this feature, also it was never used in the repo before.